facet.py

```diff
@@ -104,6 +104,7 @@ def __task_callback(self, task):
         exc = task.exception()
         if exc:
             self.__exit_point.set_exception(exc)
+        self.__tasks.discard(task)
```

Comment:
task callback called if any case when task is done, so this discard should be at the top of function, I think.

Reply:
Fixed, thank you.